core/src/test/scala/integration/kafka/admin/ReassignPartitionsIntegrationTest.scala

```diff
@@ -37,11 +83,87 @@ class ReassignPartitionsIntegrationTest extends ZooKeeperTestHarness with RackAw
 
     val topicJson = """{"topics": [{"topic": "foo"}], "version":1}"""
     val (proposedAssignment, currentAssignment) = ReassignPartitionsCommand.generateAssignment(zkClient,
-      rackInfo.keys.toSeq.sorted, topicJson, disableRackAware = false)
+      rack.keys.toSeq.sorted, topicJson, disableRackAware = false)
 
     val assignment = proposedAssignment map { case (topicPartition, replicas) =>
       (topicPartition.partition, replicas)
     }
-    checkReplicaDistribution(assignment, rackInfo, rackInfo.size, numPartitions, replicationFactor)
+    checkReplicaDistribution(assignment, rack, rack.size, numPartitions, replicationFactor)
+  }
+
+  @Test
+  def testReassignPartition(): Unit = {
```

Comment:
Seems like a basic reassignment test case. Do you think it is not covered in `ReassignPartitionsClusterTest`? It's not too clear to me why we have two separate test cases.

Reply:
It is covered. That code uses `zkClient` and `ReassignPartitionsCommand`. This test code uses AdminClient directly. Both of them test same backend code, so maybe its ok to get rid of this as when `ReassignPartitionsCommand` is fixed as part of KIP-500 we will have duplicate test.

Let me know what you prefer and I will take this test out.